core/math/a_star.cpp

```diff
@@ -487,12 +485,12 @@ void AStar::_bind_methods() {
 }
 
 AStar::AStar() {
-
+	last_max_id = -1;
 	pass = 1;
 }
 
 AStar::~AStar() {
-
+	last_max_id = -1;
```

Comment:
Unnecessary, the object would be scrapped right afterwards, and `last_max_id` is not used by `clear`. Guess the same applies to `pass`.

Reply:
This is just to leave the object in a state where it is not holding garbage data even after the destructor is called (the same pattern occurs many times throughout the godot codebase) but maybe in those cases the constructor would always be called first anyways, so it doesn't really matter?

Reply:
Removed it.